src/bittybuzz/include/bittybuzz/BittyBuzzVm.h

```diff
@@ -0,0 +1,42 @@
+#ifndef __BITTYBUZZVM_H_
+#define __BITTYBUZZVM_H_
+
+#include "bittybuzz/IBittyBuzzBytecode.h"
+#include "bittybuzz/IBittyBuzzVm.h"
+#include "bsp/IBSP.h"
+#include "logger/ILogger.h"
+#include <array>
+
+typedef struct {
+    uint8_t strId;
+    bbzvm_funp functionPtr;
+
+} FunctionRegister;
+
+class BittyBuzzVm : public IBittyBuzzVm {
+  public:
+    BittyBuzzVm(const IBittyBuzzBytecode& bytecode,
+                const IBSP& bsp,
+                const ILogger& logger,
+                const FunctionRegister* functionRegisters,
```

Comment:
Tried to find a way to pass a iterator, similar to IEnumerator in C# but it would require template, so code in the .h. suggestions are appreciated

Reply:
I mean templating is an integral part of C++ and using it requires some code in the .h so at that point I don't mind using it. It's not like if we had all of the code in the .h

Reply:
I've added a .tpp file so it's still isolated in another file for readability